sklearn/preprocessing/data.py

```diff
@@ -2348,30 +2360,39 @@ def _transform_col(self, X_col, quantiles, inverse):
             #  for inverse transform, match a uniform PDF
             X_col = output_distribution.cdf(X_col)
         # find index for lower and higher bounds
-        lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
-                            lower_bound_x)
-        upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
-                            upper_bound_x)
-
+        # comparison with NaN will raise a warning which we make silent
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
+                                lower_bound_x)
+            upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
+                                upper_bound_x)
+
+        mask_finite = np.isnan(X_col)
         if not inverse:
             # Interpolate in one direction and in the other and take the
             # mean. This is in case of repeated values in the features
             # and hence repeated quantiles
             #
             # If we don't do this, only one extreme of the duplicated is
-            # used (the upper when we do assending, and the
+            # used (the upper when we do ascending, and the
             # lower for descending). We take the mean of these two
-            X_col = .5 * (np.interp(X_col, quantiles, self.references_)
-                          - np.interp(-X_col, -quantiles[::-1],
-                                      -self.references_[::-1]))
+            X_col[~mask_finite] = .5 * (
+                np.interp(X_col[~mask_finite], quantiles, self.references_)
+                - np.interp(-X_col[~mask_finite], -quantiles[::-1],
```

Comment:
Use precomputed `X_col_finite = X_col[isfinite_mask]` instead of masking twice on the right-hand side of this assignment.

The left-hand side needs to stay `X_col[isfinite_mask] = .5 * (...`